src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.ts

```diff
@@ -160,12 +160,51 @@ export class HearingEditSummaryComponent extends RequestHearingPageFlow implemen
   }
 
   private hasHearingRequestObjectChanged(): boolean {
+    let partyDetailsModels: PartyDetailsModel[] = [];
+    let partyDetailsCompareModels: PartyDetailsModel[] = [];
+
+    if (!!this.hearingRequestMainModel?.partyDetails) {
+      partyDetailsModels = [...this.hearingRequestMainModel.partyDetails];
+      partyDetailsModels.sort(this.compareParties);
+    }
+
+    if (!!this.hearingRequestToCompareMainModel?.partyDetails) {
+      partyDetailsCompareModels = [...this.hearingRequestToCompareMainModel.partyDetails];
+      partyDetailsCompareModels.sort(this.compareParties);
+    }
+
+    const hearingRequestMainModel = {
+      requestDetails: { ...this.hearingRequestMainModel.requestDetails },
+      hearingDetails: { ...this.hearingRequestMainModel.hearingDetails },
+      caseDetails: { ...this.hearingRequestMainModel.caseDetails },
+      hearingResponse: { ...this.hearingRequestMainModel.hearingResponse },
+      partyDetails: [...partyDetailsModels]
+    };
+
+    const hearingRequestToCompareMainModel = {
+      requestDetails: { ...this.hearingRequestToCompareMainModel.requestDetails },
+      hearingDetails: { ...this.hearingRequestToCompareMainModel.hearingDetails },
+      caseDetails: { ...this.hearingRequestToCompareMainModel.caseDetails },
+      hearingResponse: { ...this.hearingRequestToCompareMainModel.hearingResponse },
+      partyDetails: [...partyDetailsCompareModels]
+    };
+
     return !_.isEqual(
-      JSON.parse(JSON.stringify(this.hearingRequestMainModel)),
-      JSON.parse(JSON.stringify(this.hearingRequestToCompareMainModel))
+      JSON.parse(JSON.stringify(hearingRequestMainModel, this.replacer)),
```

Comment:
The idea here seems to be to extract all the partyName values. This is not obvious - took me a few mins to figure it out. 
This seems like quite a big change to the way that the feature works. Previously it was comparing the whole object, now it nulls out all the values except partyName.

Is this really what is wanted. If so I suggest that you add a comment explaining why and how this works.